### PR TITLE
Tool view

### PR DESCRIPTION
**Pull request by geenen124**
Created the tool view, tool view controller, and a basic event handler structure for the tool buttons. Functionality should remain unchanged, as the mocked block creation data is still in use.

### CODE REVIEW

**Comment by Markvanderuit** [comment]
Haven't got any negative points to raise. Second member can merge.

File: src/main/java/control/ControllerManager.java [hunk 1]
```diff
@@ -18,6 +18,9 @@
     private TimelineController timelineControl;
 
     @Getter
+    private ToolViewController toolViewController;
```
Comment (bart-de-jonge) [inline src/main/java/control/ControllerManager.java:21]: Could you also add this controller to the overloaded constructor to keep things consistent. This overloaded constructor is, among others, usefull for testing.

File: src/main/java/control/ToolViewController.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        initializeTools();
+    }
+
+    /**
+     * Initialize all tools in the toolbox. Adds the corresponding buttons to the views and sets up
+     * the event handlers.
+     */
+    private void initializeTools() {
+        ToolButton blockCreationTool = new ToolButton("Add a block",
+                                                      this.controllerManager
```
Comment (bart-de-jonge) [inline src/main/java/control/ToolViewController.java:29]: You could place the last two arguments both on the same line for less lines and more readability

File: src/main/java/gui/ToolButton.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    public ToolButton(String toolName, RootHeaderArea headerArea,
+                      EventHandler<MouseEvent> clickHandler) {
+        this.name = toolName;
+        this.clickHandler = clickHandler;
+        initializeButton(headerArea);
+    }
+
+    /**
+     * Instantiates the button and adds it to the tool view.
+     * TODO: add event handler to button
```
Comment (bart-de-jonge) [inline src/main/java/gui/ToolButton.java]: The event handlers is already added to the button right?

File: src/main/java/control/ToolViewController.java (hunk 2 above)
Comment (geenen124) [inline src/main/java/control/ToolViewController.java:29]: I think that this is a more readable format, as it keeps arguments in the same vertical block. And imho line count isn't necessarily a relevant metric in this case.

File: src/main/java/control/ControllerManager.java (hunk 1 above)
Comment (geenen124) [inline src/main/java/control/ControllerManager.java:21]: Will do

File: src/main/java/control/ToolViewController.java (hunk 2 above)
Comment (bart-de-jonge) [inline src/main/java/control/ToolViewController.java:29]: I'm seeing a lot of lines with few code on it, that makes it less readable in my opinion, but that's a matter of opinion I guess

**Comment by bart-de-jonge** [comment]
Other than the comments this looks good, please fix those then we can merge

File: src/main/java/gui/ToolButton.java (hunk 3 above)
Comment (bart-de-jonge) [inline src/main/java/gui/ToolButton.java]: This isn't fixed?

**Comment by Markvanderuit** [comment]
All comments fixed. Merging.

**Pull request merged**